# **ANALISIS DATA PENJUALAN MOBIL BMW**

# **Import Library**

In [ ]:
import pandas as pd

#**Memuat Dataset**

In [ ]:
file = "/content/drive/MyDrive/AVD/bmw.csv"
df = pd.read_csv(file)
df

model  year  price transmission  mileage fuelType  tax   mpg  \
0       5 Series  2014  11200    Automatic    67068   Diesel  125  57.6   
1       6 Series  2018  27000    Automatic    14827   Petrol  145  42.8   
2       5 Series  2016  16000    Automatic    62794   Diesel  160  51.4   
3       1 Series  2017  12750    Automatic    26676   Diesel  145  72.4   
4       7 Series  2014  14500    Automatic    39554   Diesel  160  50.4   
...          ...   ...    ...          ...      ...      ...  ...   ...   
10776         X3  2016  19000    Automatic    40818   Diesel  150  54.3   
10777   5 Series  2016  14600    Automatic    42947   Diesel  125  60.1   
10778   3 Series  2017  13100       Manual    25468   Petrol  200  42.8   
10779   1 Series  2014   9930    Automatic    45000   Diesel   30  64.2   
10780         X1  2017  15981    Automatic    59432   Diesel  125  57.6   

       engineSize  
0             2.0  
1             2.0  
2             3.0  
3             1.5  
4             3.0  
...           ...  
10776         2.0  
10777         2.0  
10778         2.0  
10779         2.0  
10780         2.0  

[10781 rows x 9 columns]

In [ ]:
df.head()

model  year  price transmission  mileage fuelType  tax   mpg  \
0   5 Series  2014  11200    Automatic    67068   Diesel  125  57.6   
1   6 Series  2018  27000    Automatic    14827   Petrol  145  42.8   
2   5 Series  2016  16000    Automatic    62794   Diesel  160  51.4   
3   1 Series  2017  12750    Automatic    26676   Diesel  145  72.4   
4   7 Series  2014  14500    Automatic    39554   Diesel  160  50.4   

   engineSize  
0         2.0  
1         2.0  
2         3.0  
3         1.5  
4         3.0

In [ ]:
df.tail()

model  year  price transmission  mileage fuelType  tax   mpg  \
10776         X3  2016  19000    Automatic    40818   Diesel  150  54.3   
10777   5 Series  2016  14600    Automatic    42947   Diesel  125  60.1   
10778   3 Series  2017  13100       Manual    25468   Petrol  200  42.8   
10779   1 Series  2014   9930    Automatic    45000   Diesel   30  64.2   
10780         X1  2017  15981    Automatic    59432   Diesel  125  57.6   

       engineSize  
10776         2.0  
10777         2.0  
10778         2.0  
10779         2.0  
10780         2.0

# **Business Understanding**

## **Business Objective**

Dataset ini berisi sekumpulan data tentang mobil bekas BMW. Data ini memiliki variabel utama yaitu Model, Tahun Keluaran Mobil, Jarak tempuh, Jenis Bahan Bakar, Jenis Transmisi dan model mobil. Tujuan dari menganalisis data ini adalah:
*   Mengetahui Faktor-Faktor apa saja yang akan menentukan harga sebuah mobil BMW bekas
*   Membantu Dealer untuk menentukan harga jual mobil bekas BMW agar masuk akal dan tetap laku dipasaran.
*   Mengidentifikasi tren di pasaran. Contohnya adalah "apakah mobil diesel lebih laku? atau apakah mobil matic lebih mahal?"

Kesimpulan:
Tujuan utama data ini untuk memudahkan dealer dalam memprediksi dan menentukan harga jual mobil BMW bekas yang masuk akal dan kompetitif, serta memprediksi dan menentukan tren pasar BMW sehingga pengadaan mobil bekas BMW bisa lebih sesusai dengan kebutuhan pasar.

## **Assess Situation**


Dataset ini menampung data yang cukup besar yaitu 10781 baris sehingga akan memperkuat prediksi harga dan faktor yang mempengaruhi harga tersebut, karena semakin banyak data yang dianalisis, maka akan semakin akurat. Selain itu berdasarkan analisis Mayoritas mobil diproduksi sekitar tahun 2017 dan Kapasitas Mesin rata-rata 2.0L. Selain itu tidak ditemukan nilai pada baris data yang hilang di salah satu kolom variabel, sehingga prediksi bisa semakin akurat dikarenakan lengkapnya data didalam dataset ini.

Untuk situasi bisnis, pasar mobil bekas biasanya dipengaruhi oleh:
*   Semakin baru tahun mobil maka akan semakin tingi harga jualnya.
*   Semakin tinggi jarak tempuh maka harga akan semakin turun
*   Model mobil akan mempengaruhi harga mobil. Misalnya model yang premium seperti 5 series kemungkinan harga jualnya akan mahal.
*   Mobil dengan transmisi otomatis (mobil matic) biasanya lebih mahal dibanding mobil dengan transmisi manual.
*   Mobil dengan bahan bakar diesel bisa lebih murah atau mahal, tergantung target dan pasarnya.
*  faktor-faktor atau variabel diluar dataset ini bisa jadi akan mempengaruhi harga jual mobil, misalnya kondisi fisik, lokasi, warna, dsb

Kesimpulannya adalah menganalisis dataset ini akan mendukung prediksi harga jual mobil bekas BMW yang lebih akurat karena faktor atau variabel yang dianalisis dalam dataset ini adalah faktor-faktor yang memang menjadi pertimbangan dalam situasi bisnis pasar mobil bekas.


## **Analytic Goals Dan Project Plan**


### **Analytic Goals**

Tujuan utama dari analisis data ini adalah Mengidentifikasi Faktor yang paling memengaruhi harga mobil, menganalisis hubungan antara:

*   Tahun dengan Harga Jual
*   Jarak tempuh dengan Harga Jual
*   Jenis Bahan Bakar dengan Harga
*   jenis transmisi dengan harga
*   Model Mobil dengan harga Jual

Mengidentifikasi tren pasar yaitu:
*   Mengidentifikasi model yang memiliki rata-rata harga tertinggi.
*   Menganalisis distribusi jenis bahan bakar dalam dataset.
*   menganalisis distribusi harga antara transmisi otomatis (matic) dengan manual

Dari sisi perusahaan ini akan menguntungkan dalam:
*   membantu perusahaan dalam mengindentifikasi tren Pasar dan faktor penentu harga jual mobil
*   Menghasilkan visualisasi
*   menyediakan dasar pengambilan keputusan




### **Project Plan**


**1. Data Understanding**
Pada tahap ini akan dilakukan:
*   Memahami struktur dataset (jumlah baris dan kolom)
*   Memahami variabel utama yang nantinya akan dijadikan faktor penentuan harga jual mobil bekas BMW seperti model, jarak tempu, harga jual, dst.
*   Mengecek apakah terdapat data kosong (missing value)
*   Mengidentifikasi kemungkinan adanya data yang tidak wajar pada variabel seperti price dan mileage

**2. Data Preparation**
Tahap ini dilakukan untuk menyiapkan data sebelum dianalisis lebih lanjut.

Langkah yang dilakukan adalah:
*   Membersihkan data jika ditemukan nilai yang tidak wajar
*   Mengubah variabel kategorikal (model, fuel type, transmission) menjadi format yang bisa dianalisis jika akan dilakukan modeling
*   Menyederhanakan atau mengelompokkan data jika diperlukan

**3. Exploratory Data Analysis (EDA)**
Pada tahap ini dilakukan analisis dan visualisasi untuk menjawab analytic goals, seperti:
*   Menganalisis hubungan antara tahun mobil dan dengan harga
*   Menganalisis hubungan antara jarak tempuh dan harga
*   Membandingkan rata-rata harga berdasarkan:
    - model
    - Jenis Bahan Bakar
    - Transmisi
*   Membuat visualisasi seperti:
    - Scatter plot
    - Bar chart
    - Boxplot
    - Distribusi histogram
*   Membuat Dashboard












